src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -133,15 +132,15 @@ private void CreateStreamAndStartWatcher()
             if (String.IsNullOrEmpty(_directory) == false)
             {
                 // Get the path to watch and verify we created the CFStringRef
-                CFStringRef path = Interop.CoreFoundation.CFStringCreateWithCString(_directory);
-                if (path == IntPtr.Zero)
+                SafeCreateHandle path = Interop.CoreFoundation.CFStringCreateWithCString(_directory);
+                if (path.IsInvalid)
                 {
                     throw Interop.GetExceptionForIoErrno(Marshal.GetLastWin32Error(), _directory, true);
                 }
 
                 // Take the CFStringRef and put it into an array to pass to the EventStream
-                CFArrayRef arrPaths = Interop.CoreFoundation.CFArrayCreate(new CFStringRef[1] { path }, 1);
-                if (arrPaths == IntPtr.Zero)
+                SafeCreateHandle arrPaths = Interop.CoreFoundation.CFArrayCreate(new CFStringRef[1] { path.DangerousGetHandle() }, 1);
+                if (arrPaths.IsInvalid)
```

Comment:
Their finalizers will eventually clean them up.  But there is cost to that, and the resources will end up getting released at some non-deterministic point in the future.  While that's a fine fallback, in general if it's easy to proactively clean things up, we should.